Bottom: 4ec6a22985dd555daefe54c655ac80792102a474
Top:    b8ad80b7551b81a96bff487fd8005f8ec859f33e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-23 21:40:11 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index cd2aff5..c07c795 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2618,7 +2618,12 @@ IsPolygonInPolygon (PolygonTypePtr P1, PolygonTypePtr P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
